django/db/models/sql/query.py

```diff
@@ -1000,9 +1000,12 @@ def prepare_lookup_value(self, value, lookups, can_reuse, allow_joins=True):
         # Interpret '__exact=None' as the sql 'is NULL'; otherwise, reject all
         # uses of None as a query value.
         if value is None:
-            if lookups[-1] not in ('exact', 'iexact'):
-                raise ValueError("Cannot use None as a query value")
-            return True, ['isnull'], used_joins
+            if lookups[-1] not in self.query_terms:
```

Comment:
I don't know time to look at this in detail right now but maybe https://github.com/django/django/pull/8704#issuecomment-314124887 helps.